explorer/websockethandlers.go

```diff
@@ -59,55 +60,108 @@ func (exp *explorerUI) RootWebsocket(w http.ResponseWriter, r *http.Request) {
-					if len(msg.Message) > requestLimit {
-						log.Debug("Request size over limit")
-						webData.Message = "Request too large"
-						break
-					}
 					log.Debugf("Received decodetx signal for hex: %.40s...", msg.Message)
 					tx, err := exp.blockData.DecodeRawTransaction(msg.Message)
 					if err == nil {
 						message, err := json.MarshalIndent(tx, "", "    ")
 						if err != nil {
 							log.Warn("Invalid JSON message: ", err)
-							webData.Message = fmt.Sprintf("Error: Could not encode JSON message")
+							webData.Message = "Error: Could not encode JSON message"
 							break
 						}
 						webData.Message = string(message)
 					} else {
 						log.Debugf("Could not decode raw tx")
 						webData.Message = fmt.Sprintf("Error: %v", err)
 					}
+
 				case "sendtx":
-					webData.EventId = msg.EventId + "Resp"
-					if len(msg.Message) > requestLimit {
-						log.Debugf("Request size over limit")
-						webData.Message = "Request too large"
-						break
-					}
 					log.Debugf("Received sendtx signal for hex: %.40s...", msg.Message)
 					txid, err := exp.blockData.SendRawTransaction(msg.Message)
 					if err != nil {
 						webData.Message = fmt.Sprintf("Error: %v", err)
 					} else {
 						webData.Message = fmt.Sprintf("Transaction sent: %s", txid)
 					}
-				case "getmempooltxs":
-					webData.EventId = msg.EventId + "Resp"
 
-					exp.MempoolData.Lock()
+				case "getmempooltxs":
+					exp.MempoolData.RLock()
 					msg, err := json.Marshal(exp.MempoolData)
-					exp.MempoolData.Unlock()
+					exp.MempoolData.RUnlock()
+
+					if err != nil {
+						log.Warn("Invalid JSON message: ", err)
+						webData.Message = "Error: Could not encode JSON message"
+						break
+					}
+					webData.Message = string(msg)
+
+				case "getticketpooldata":
+					// Ensures that multiple requests for the same update are
+					// treated as the single request.
+					if !tpEvent.Active {
+						log.Warn("ticketpool event has not updated data.")
+						break
+					}
+
+					tpEvent.Active = false
+
+					cData, gData, err := exp.explorerSource.TicketPoolVisualization(
+						dbtypes.ChartGroupingFromStr(msg.Message),
+					)
+					if err != nil {
+						if strings.HasPrefix(err.Error(), "unknown interval") {
+							log.Debugf("Invalid ticket pool interval provided via getticketpooldata: %s", msg.Message)
+							webData.Message = "Error: " + err.Error()
+							break
+						}
+						log.Errorf("TicketPoolVisualization error: %v", err)
+						webData.Message = "Error: Failed to fetch ticketpool data"
+						break
+					}
+
+					exp.MempoolData.RLock()
+					var mpData = exp.MempoolData
+
+					var mp = dbtypes.PoolTicketsData{}
 
+					if len(mpData.Tickets) > 0 {
+						mp.Time = append(mp.Time, uint64(mpData.Tickets[0].Time))
+						mp.Price = append(mp.Price, mpData.Tickets[0].TotalOut)
+						mp.Mempool = append(mp.Mempool, uint64(len(mpData.Tickets)))
+					} else {
+						log.Debug("No tickets exists in the mempool")
+					}
+
```

Comment:
You could put `exp.MempoolData.RUnlock()` here since you are done with `mpData`, but it's not a big difference since only a small struct definition follows.

Reply:
ok. Noted